zingolib/src/lightclient.rs

```diff
@@ -260,6 +260,45 @@ impl LightClient {
     }
 }
 impl LightClient {
+    /// The wallet this fn associates with the lightclient is specifically derived from
+    /// a spend authority.
+    pub async fn new_from_wallet_base_async(
+        wallet_base: WalletBase,
+        config: &ZingoConfig,
+        birthday: u64,
+        overwrite: bool,
+    ) -> io::Result<Self> {
+        #[cfg(not(any(target_os = "ios", target_os = "android")))]
+        {
+            if !overwrite && config.wallet_exists() {
+                return Err(Error::new(
```

Comment:
As with `create_zingoconfdir_async`, this should be straightforward to dry.

Reply:
DRYed